data/processors/nat.py

```diff
@@ -0,0 +1,195 @@
+import json
+import logging
+from collections import Counter
+from dataclasses import dataclass
+
+
+@dataclass
+class NATBuilding:
+    b_id: None | str
+    b_code: str
+    b_name: str
+    b_tumonline_id: None | int
+    b_alias: None | str
+    b_address: None | str
+
+    def __init__(self, data: dict):
+        self.b_id = None  # Later set by _infer_internal_id()
+        self.b_code = data["building_code"]  # Building id/code used by the NAT roomfinder
+        self.b_name = data["building_name"]
+        self.b_tumonline_id = data["building_id"]
+        self.b_alias = data["building_short"]
+        self.b_address = data["address"]
+
+    def as_dict(self):
+        """Return the building data as dict"""
+        return self.__dict__
+
+
+def merge_nat_buildings(data):
+    """
+    Merge the buildings in the NAT Roomfinder with the existing data.
+    This may overwrite existing data, if they have patched some fields.
+    """
+    with open("external/results/buildings_nat.json", encoding="utf-8") as file:
+        buildings = json.load(file)
+
+    # Sanity-check: Make sure that the buildings in the data are unique
+    building_ids = [b["building_code"] for b in buildings]
+    duplicate_building_ids = {b_id: cnt for b_id, cnt in Counter(building_ids).items() if cnt > 1}
+    if duplicate_building_ids:
+        raise ValueError(f"There are duplicate buildings in the data: {duplicate_building_ids}")
+
+    for building in [NATBuilding(b) for b in buildings]:
+        if building.b_code in [
```

Comment:
Given that this list has grown considerable, extracting this to a configuration file is probably wise

Reply:
I've moved it to a YAML file